app/questionnaire/questionnaire_manager.py

```diff
@@ -149,9 +149,12 @@ def _get_next_answer_instance(answer_store, answer_id):
         return next_answer_instance_id
 
     def remove_answer(self, location, answer_store, index_to_remove):
-        answer = self.state.get_answers()[int(index_to_remove)]
-        question = answer.parent
-        question.remove_answer(answer)
 
-        answer_store.remove(answer.flatten())
+        state_answers = self.state.get_answers()
+        for state_answer in state_answers:
+            if state_answer.answer_instance == index_to_remove:
```

Comment:
I may be wrong here. As `state_answers = self.state.get_answers()` will only return answers for the current block not for all answers for the whole questionnaire